pkg/autoupgrade/daemon.go

```diff
@@ -206,9 +206,11 @@ func (d *daemon) refreshImages(ctx context.Context, apps map[kclient.ObjectKey]v
 			// Updated can be false for two reasons:
 			// 1. The tag was a pattern and a newer tag was not found
 			// 2. The tag was not a pattern
-			// In either case, we also want to check to see if new content was pushed to the current tag
+			// In either case, we also want to check to see if new content was pushed to the current tag, if the image has a current tag.
 			// This satisfies the usecase of autoUpgrade with an app's tag is something static, like "latest"
-			if !updated {
+			// However, if the tag is a pattern and the current image has no tag, we don't want to check for a digest because this would
+			// result in a digest upgrade even though no tag matched.
+			if !updated && (!isPattern || current.Identifier() != "") {
```

Comment:
If I have a pattern, say `v#.#` and the app is currently on `v1.0` and I push a new version of `v1.0`, will the app get updated?

Reply:
Yes, because `current.Identifier() != ""` in that case.